cosmetics-web/spec/models/nano_element_spec.rb

```diff
@@ -28,6 +28,60 @@
         end
       end
     end
+
+    context "when using same name in same notification" do
+      let(:notification) { create(:notification) }
+      let(:existing_name) { "Nanomaterial" }
+      let(:new_name) { existing_name }
+
+      let(:nano_material1) { create(:nano_material, notification: notification) }
+      let(:nano_material2) { create(:nano_material, notification: notification) }
+
+      let(:nano_element1) { create(:nano_element, inci_name: existing_name, nano_material: nano_material1) }
+      let(:nano_element) { build(:nano_element, inci_name: new_name, nano_material: nano_material2) }
+
+      before do
+        nano_element1
+      end
+
+      it "is valid with same name without context" do
+        nano_element.valid?
+        expect(nano_element.errors).to be_empty
+      end
+
+      it "is valid with same name with context" do
+        nano_element.valid?(:add_nanomaterial_name)
+        expect(nano_element.errors).to be_present
+      end
+
+      context "when saving" do
+        # rubocop:disable RSpec/PredicateMatcher
+        it "does not cause error on self" do
+          expect(nano_element1.valid?(:add_nanomaterial_name)).to be_truthy
+        end
+        # rubocop:enable RSpec/PredicateMatcher
```

Comment:
Suggestion to avoid having to disable the Rubocop rule:
```suggestion
        it "does not cause error on self" do
          expect(nano_element1.valid?(:add_nanomaterial_name)).to be(true)
        end
```